core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -114,6 +114,9 @@ object Defaults {
   val LogCleanerDeleteRetentionMs = 24 * 60 * 60 * 1000L
   val LogCleanerMinCompactionLagMs = 0L
   val LogCleanerMaxCompactionLagMs = Long.MaxValue
+  val LogCleanerCompactionStrategyOffset = "offset"
```

Comment:
The possible values are defined in both TopicConfig for per-topic overrides as well as where for global configuration. I'd suggest we just define it in one place, .e.g defining the possible values in `TopicConfig`, similar to the viable values of `cleanup.policy` and then in KafkaConfig here we just refer to the values of TopicConfig.